src/OpenTK.Windowing/NativeWindow.cs

```diff
@@ -62,60 +57,241 @@ public bool Visible
+        }
+        public int Width
+        {
+            get { return Size.Width; }
+            set { Size = new Size(value, Size.Height); }
+        }
+        public int Height
+        {
+            get { return Size.Height; }
+            set { Size = new Size(Size.Width, value); }
+        }
+        public Rectangle ClientRectangle { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
 
-        public IWindowInfo WindowInfo => throw new NotImplementedException();
+        public Size ClientSize
+        {
+            get
+            {
+                int width, height;
+                unsafe { glfw.GetFramebufferSize(windowPtr, out width, out height); }
+                return new Size(width, height);
+            }
+            set => throw new NotImplementedException();
+        }
 
-        public WindowState WindowState { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public WindowBorder WindowBorder { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Rectangle Bounds { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Point Location { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Size Size { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int X { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Y { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Width { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Height { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Rectangle ClientRectangle { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Size ClientSize { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public MouseCursor Cursor { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public bool CursorVisible { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public bool CursorGrabbed { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public MouseCursor Cursor
+        {
+            get
+            {
+                throw new NotImplementedException();
 
+            }
+            set => throw new NotImplementedException();
+        }
 
-        public NativeWindow(int width = 640, int height = 480, string title = "", bool windowedFullscreen = false)
+        private bool _cursorHidden;
```

Comment:
perhaps naming incosistency between "_cursorHidden" and "CursorVisible" as they should be the complete opposite of each other?

Reply:
The names are similar but they are very distinct fields. _cursorHidden is a stop-gap to account for the cursor being able to be grabbed _and_ hidden. if we set CursorGrabbed to false. it has to know that we need to set `InputModeValue.CursorHidden` and not `InputModeValue.CursorNormal`

It's a naive solution to preserve the existing API, let me know if something better comes to mind.

Reply:
Fixed by latest commit